### PR TITLE
SWATCH-658: Internal metering API should not support accountNumber.

### PR DESCRIPTION
**Pull request by vbusch**
Jira issue: [SWATCH-658](https://issues.redhat.com/browse/SWATCH-658)

## Description
From Internal metering API, Stop accepting accountNumber and orgId should be marked as required.

Prior to this PR, it was possible to leave both the accountNumber and orgID empty.  It would then try to find all events for the provided range.  However, that feature wasn't currently working and would later throw an exception for the missing orgId.  This PR makes orgId mandatory.

### Setup
1. To run locally prometheus by pointing to stage follow the steps in: [Connecting  Observatorium to fetch Prometheus ](https://docs.engineering.redhat.com/display/ENT/Connecting+to+Observatorium+to+Fetch+Prometheus+Metrics)

* Copy CLIENT_SECRET and paste it in a file `token-refresher`
* Run: `podman run --name=telemeter-stage --rm -ti -p 8082:8080     quay.io/observatorium/token-refresher:master-2021-02-05-5da9663     --oidc.client-secret=$(cat ~/Downloads/token-refresher)     --oidc.client-id=observatorium-subwatch-staging     --oidc.issuer-url=https://sso.redhat.com/auth/realms/redhat-external     --url=https://observatorium.api.stage.openshift.com`

2. Apply the below patch in `application-openshift-metering-worker.yaml` since billing_model="marketplace" won't give results in prometheus in the further steps

```
          diff --git a/src/main/resources/application-openshift-metering-worker.yaml b/src/main/resources/application-openshift-metering-worker.yaml
--- a/src/main/resources/application-openshift-metering-worker.yaml	(revision ce8b242be21cb1b8329d08cfd46f684065e44126)
+++ b/src/main/resources/application-openshift-metering-worker.yaml	(date 1689775711055)
@@ -12,7 +12,7 @@
           default: >-
             #{metric.queryParams[prometheusMetric]}
             * on(_id) group_right
-            min_over_time(#{metric.queryParams[prometheusMetadataMetric]}{product="#{metric.queryParams[product]}", external_organization="#{runtime[orgId]}", billing_model="marketplace", support=~"Premium|Standard|Self-Support|None"}[1h])
+            min_over_time(#{metric.queryParams[prometheusMetadataMetric]}{product="#{metric.queryParams[product]}", external_organization="#{runtime[orgId]}", support=~"Premium|Standard|Self-Support|None"}[1h])
           addonSamples: >-
             #{metric.queryParams[prometheusMetric]}
             * on(_id) group_right
```

3. Then in separate terminal pointing to the prom_url in the doc run:  `RHSM_SUBSCRIPTIONS_ENABLE_SYNCHRONOUS_OPERATIONS=true DEV_MODE=true PROM_URL=http://localhost:8082/api/metrics/v1/telemeter/api/v1 ./gradlew clean :bootRun`
4. Go to prometheus url that points to stage from local: `http://localhost:8082/api/metrics/v1/telemeter/graph`
   Under graph tab paste the query `cluster:usage:workload:capacity_physical_instance_hours * on(_id) group_right min_over_time(subscription_labels{product="osd", external_organization="6323660", support=~"Premium|Standard|Self-Support|None"}[1h])`

* or click on this [link](http://localhost:8082/api/metrics/v1/telemeter/graph?g0.expr=cluster%3Ausage%3Aworkload%3Acapacity_physical_instance_hours%20*%20on(_id)%20group_right%20min_over_time(subscription_labels%7Bproduct%3D%22osd%22%2C%20external_organization%3D%226323660%22%2C%20support%3D~%22Premium%7CStandard%7CSelf-Support%7CNone%22%7D%5B1h%5D)&g0.tab=0&g0.stacked=0&g0.range_input=2h&g0.max_source_resolution=0s&g0.deduplicate=1&g0.partial_response=0&g0.store_matches=%5B%5D&g0.end_input=2023-08-08%2004%3A49%3A17&g0.moment_input=2023-08-08%2004%3A49%3A17) and adjust the timestamps but this is only applicable locally.

### Steps
1. Run: `curl -X POST  -H "x-rh-swatch-psk: placeholder" "http://localhost:8000/api/rhsm-subscriptions/v1/internal/metering/OpenShift-dedicated-metrics?orgId=6323660&rangeInMinutes=1200"`

### Verification
1. Depending on if stage prometheus has data you should see bunch of inserts in events table with new event_types in event_type column and in data json blob like below:
   `INSERT INTO public.events (id, account_number, timestamp, data, event_type, event_source, instance_id, org_id) VALUES ('743b86eb-56fa-4b2a-84c0-a243fb844a87', '1455657', '2023-08-08 10:00:00.000000 +00:00', '{"sla": "Premium", "role": "osd", "org_id": "6323660", "event_id": "743b86eb-56fa-4b2a-84c0-a243fb844a87", "timestamp": "2023-08-08T10:00:00Z", "event_type": "snapshot_openshift-dedicated-metrics_cores", "expiration": "2023-08-08T11:00:00Z", "instance_id": "fbe99a5e-9213-4c10-942a-74b7df181ec2", "display_name": "fbe99a5e-9213-4c10-942a-74b7df181ec2", "event_source": "prometheus", "measurements": [{"uom": "Cores", "value": 4.666666666666667}], "service_type": "OpenShift Cluster", "account_number": "1455657", "billing_provider": "red hat", "billing_account_id": null}', 'snapshot_openshift-dedicated-metrics_instance-hours', 'prometheus', 'fbe99a5e-9213-4c10-942a-74b7df181ec2', '6323660'); `

**In logs:** `2023-08-08 18:01:13,974 [thread=http-nio-8000-exec-5] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.PrometheusMeteringController] self- Persisted 77 events for OpenShift-dedicated-metrics Instance-hours metrics. `

### CODE REVIEW

**Comment by barnabycourt** [comment]
ok to test

**Review by kartikshahc — CHANGES_REQUESTED** [review]

File: src/main/java/org/candlepin/subscriptions/metering/api/admin/InternalMeteringResource.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 import io.micrometer.core.annotation.Timed;
 import jakarta.transaction.Transactional;
 import jakarta.validation.constraints.Min;
-import jakarta.ws.rs.BadRequestException;
+import jakarta.ws.rs.*;
```
[the remark this answers is not in the document]
Comment (vbusch) [reply src/main/java/org/candlepin/subscriptions/metering/api/admin/InternalMeteringResource.java]: I have fixed this and modified my settings.

File: src/main/java/org/candlepin/subscriptions/metering/service/prometheus/task/PrometheusMetricsTaskManager.java [hunk 2]
```diff
@@ -113,13 +113,6 @@ private void queueMetricUpdateForOrgId(
...
 
-  @Transactional
-  public void updateMetricsForAllAccounts(
-      String productTag, OffsetDateTime start, OffsetDateTime end) {
-    updateMetricsForAllAccounts(
-        productTag, start, end, RetryTemplate.builder().maxAttempts(1).build());
-  }
-
   @Transactional
   public void updateMetricsForAllAccounts(String productTag, int rangeInMinutes) {
```
[the remark this answers is not in the document]
Comment (vbusch) [reply src/main/java/org/candlepin/subscriptions/metering/service/prometheus/task/PrometheusMetricsTaskManager.java]: Done :-)

**Comment by Aurobinda55** [comment]
- Added an event in the local Prometheus and perform metering without orgId and got below error 

` self- SUBSCRIPTIONS1002: Client request failed validation. - meterProductForOrgIdAndRange.orgId: must not be null`

- Added an event with org_id = 14541308 and perform metering for 14492399 and got the below logs 
`Persisted 0 events for OpenShift-metrics Cores metrics for org 14492399 `
 that means it trying to perform metering based on org_id and not the events in a particular range 
- Added one event with org_id=14541308 and perform metering for 14541308 and got below successful logs

```
07:48:36,214 [thread=metering-task-processor-9-C-1] [INFO ] [org.candlepin.subscriptions.metering.task.MetricsTask] - Running rhacs Cores metrics update task for orgId: 14541308
2023-08-11 07:48:36,218 [thread=metering-task-processor-9-C-1] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.PrometheusMeteringController] - Collecting metrics for orgId=14541308: rhacs Cores
2023-08-11 07:48:36,218 [thread=metering-task-processor-9-C-1] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.PrometheusService] - Fetching metrics from prometheus: 2023-08-10T00:00Z -> 2023-08-11T23:00Z [Step: 3600]
2023-08-11 07:48:36,241 [thread=metering-task-processor-9-C-1] [INFO ] [org.candlepin.subscriptions.metering.service.prometheus.PrometheusMeteringController] - Persisted 24 events for rhacs Cores metrics.
2023-08-11 07:48:36,243 [thread=metering-task-processor-9-C-1] [INFO ] [org.candlepin.subscriptions.metering.task.MetricsTask] - rhacs Cores metrics task complete.
```

**Review by kartikshahc — APPROVED** [review]

**Pull request merged**